plugin-java/che-plugin-java-ext-jdt/org-eclipse-jdt-ui/src/main/java/org/eclipse/jdt/internal/core/JavaModel.java

```diff
@@ -203,10 +202,23 @@ public IJavaProject getJavaProject(IResource resource) {
  * @see org.eclipse.jdt.core.IJavaModel
  */
 public IJavaProject[] getJavaProjects() throws JavaModelException {
-    ArrayList list = getChildrenOfType(IJavaElement.JAVA_PROJECT);
-    IJavaProject[] array = new IJavaProject[list.size()];
-    list.toArray(array);
-    return array;
+//    ArrayList list = getChildrenOfType(IJavaElement.JAVA_PROJECT);
+	// determine my children
+	IProject[] projects = ResourcesPlugin.getWorkspace().getRoot().getProjects();
+	int length = projects.length;
+	IJavaProject[] children = new IJavaProject[length];
+	int index = 0;
+	for (int i = 0; i < length; i++) {
+		IProject project = projects[i];
+		if (org.eclipse.jdt.internal.core.JavaProject.hasJavaNature(project)) {
+			children[index++] = getJavaProject(project);
+		}
+	}
+	if (index < length)
+		System.arraycopy(children, 0, children = new IJavaProject[index], 0, index);
+//    IJavaProject[] array = new IJavaProject[list.size()];
```

Comment:
no, it's eclipse code, https://jira.codenvycorp.com/browse/IDEX-3609 after this I return commented code